src/linked_list.rs

```diff
@@ -234,6 +234,27 @@ impl<'a, A: Adapter<Link = Link>> Cursor<'a, A> {
         }
     }
 
+    /// Clones and returns the pointer that points to the element that the
+    /// cursor is referencing.
+    ///
+    /// This returns None if the cursor is currently pointing to the null
+    /// object.
+    #[inline]
+    pub fn get_pointer(&self) -> Option<A::Pointer>
+    where
+        A::Pointer: Clone,
+    {
+        if self.is_null() {
+            None
+        } else {
+            let pointer = unsafe {
+                let value = &*self.list.adapter.get_value(self.current.0);
+                core::mem::ManuallyDrop::new(A::Pointer::from_raw(value as *const A::Value))
```

Comment:
I think that a more correct implementation here would be to call `into_raw` after cloning rather than using a `ManuallyDrop`. Note that you will need to do this with a drop guard, so that `into_raw` is called even if `clone` panics.

Reply:
I implemented it now with a drop guard. It's a  bit on the ugly side, since the intrusive pointer needs to be stored in an `Option` - otherwise we can't pass it to `into_raw` in `drop` (can not move out of the struct). However I at least moved everything into a helper function, so everything is in one place now.